### DISCUSSION TITLE
Detection and/or Responsive?

### DISCUSSION THREAD

**Discussion by wangkanai** [wangkanai/wangkanai#343] [Ideas]
Historically `Responsive` was a separate package that you needed to install which had dependency on `Detection`, then during my development upto [3.0-alpha08](https://github.com/wangkanai/Detection/releases/tag/3.0-Alpha08). `Responsive` was bake into `Detection` as a basic service. If you don't want to use Responsive, [3.0-alpha03](https://github.com/wangkanai/Detection/releases/tag/3.0-Alpha03) you can disable Responsive #247 via `DetectionOptions`. Another question arise now, the Session and Responsive services was still registered. Responsive require Session because it has a feature added during [3.0-alpha04](https://github.com/wangkanai/Detection/releases/tag/3.0-Alpha04) User Preference #236.

`Detection` only resolve the [User-Agent](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent) given from the web browser. Which is very light weight for the ASP.NET Core Web Application. While `Responsive` has advance features, it is more intensive. If you don't need, and only need `Detection`. `Responsive` service and its dependency are still loaded into the application memory at runtime, even its was disabled via the option.

Another issue with `Responsive` that is `Session` would create a Cookie on the client web browser. Its would cause [GDPR](https://gdpr-info.eu/) concern if it effect you or want to avoid the issue. Reference [GDPR](https://gdpr-info.eu/) which I don't have time to look into myself. If GDPR also include identifying the web visitor via [User-Agent](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent) then `Detection` is effected. Therefore, `Responsive` would be subjected GDPR by its require dependency. Make this issue `Responsive` only a software architecture concern. 

**Therefore, the question is that `Responsive` service should included when you `services.AddDetection()` or `Responsive` should be manually added via `services.AddResponsive()`?** This is while `Responsive` is still in the same library package.

**Comment by wangkanai** [comment]
What is your ideas and opinions? 

@vovikdrg @rynowak @KostaVlev @TheDarkCode @monkeywithacupcake  and anybody effect by this issue.

**Comment by vovikdrg** [comment]
What is performance impact on this?

**Comment by wangkanai** [comment]
@vovikdrg I don't know and not sure how to measure it. I just realize architecturally that its doesn't sound right. 

**Edit**

Come to think of it, Maybe a load stress test could answer this question. I will put it on my todo list.

**Comment by wangkanai** [comment]
#369 PR help answer the question about Detection with/out Responsive.

Thank you very much @dv00d00 for your help.

**Comment by wangkanai** [comment]
Move to disccusion #476

**Comment by wangkanai** [comment]
Final conclusion, Responsive has now been separated from Detection in 5.0
